### ISSUE TITLE
Postmark metadata key names longer than 20 chars cause silent send failure

### ISSUE THREAD

**Issue by chrisgrande** [anymail/django-anymail#238]
* Anymail version **8.2**
* ESP (Mailgun, SendGrid, etc.) **Postmark**
* Versions of Django, requests, python **Django 3.2.3, Requests 2.25.1, Python 3.9.2**

Using the example code from the docs:
```
from anymail.message import AnymailMessage

message = AnymailMessage(
    subject="Welcome",
    body="Welcome to our site",
    to=["New User <user1@example.com>"],
    tags=["Onboarding"],  # Anymail extra in constructor
)
# Anymail extra attributes:
message.metadata = {"onboarding_experiment": "variation 1"}
message.track_clicks = True

message.send()
status = message.anymail_status
```

Sending via Postmark will result in a silent failure, with the `anymail_status` returning as `unknown` with no other errors.

I tracked the issue down to `message.metadata = {"onboarding_experiment": "variation 1"}`, and the key name containing a hyphen or underscore. `onboarding_experiment` or `onboarding-experiment` will fail, however `onboardingexperiment` will succeed.

**Comment by medmunds** [comment]
Thanks for the report and including the exact test case that shows the issue.

When I run your test with `"DEBUG_API_REQUESTS": True`, I see Postmark is returning an error. The problem is not hyphens or underscores, but the length of the metadata key name:

```text
HTTP 422 Unprocessable Entity

{"ErrorCode":300,"Message":"Invalid metadata content. Field names
 are limited to 20 characters and field values are limited to 80 characters
 in length. The 'onboarding_experiment' field in your metadata content
 exceeds these limits."}
```

It took me a while to figure out why Anymail isn't raising an error for this case. There's a bug in how Anymail's Postmark backend is handling ErrorCode 300 errors. Because Postmark reuses this code for several different API errors, Anymail has to resort to parsing the human-readable Message field to recognize certain recipient-related errors it needs to handle differently. (And it needs to handle those recipient-related errors differently in order to match how Django's default SMTP EmailBackend handles refused recipients. Sigh.)

Fix coming shortly.

**Comment by medmunds** [comment]
Fix released in v8.3

**Comment by chrisgrande** [comment]
Awesome @medmunds — thank you for the quick fix, and for mentioning `DEBUG_API_REQUESTS`, I had completely missed that setting! What are the odds on `onboarding_experiment` being exactly 21 characters 🤣!
